File: apps/mobile/lib/core/models/citizen.dart
class Citizen {
  final String id;
  final String firstNameAr;
  final String? fatherNameAr;
  final String? grandfatherNameAr;
  final String familyNameAr;
  final String? phone;
  final String? legacyNationalNo;
  final int? regionId;
  final String? digitalIdNumber; // active card if present
  final String? photoPath;

  Citizen({
    required this.id,
    required this.firstNameAr,
    this.fatherNameAr,
    this.grandfatherNameAr,
    required this.familyNameAr,
    this.phone,
    this.legacyNationalNo,
    this.regionId,
    this.digitalIdNumber,
    this.photoPath,
  });

  String get fullNameAr => [
        firstNameAr,
        fatherNameAr ?? '',
        grandfatherNameAr ?? '',
        familyNameAr,
      ].where((p) => p.isNotEmpty).join(' ');

  factory Citizen.fromJson(Map<String, dynamic> json) => Citizen(
        id: json['id'] as String,
        firstNameAr: json['first_name_ar'] as String? ?? '',
        fatherNameAr: json['father_name_ar'] as String?,
        grandfatherNameAr: json['grandfather_name_ar'] as String?,
        familyNameAr: json['family_name_ar'] as String? ?? '',
        phone: json['phone'] as String?,
        legacyNationalNo: json['legacy_national_no'] as String?,
        regionId: (json['region_id'] as num?)?.toInt(),
        digitalIdNumber: json['digital_id_number'] as String?,
        photoPath: json['photo_path'] as String?,
      );

  Citizen copyWith({
    String? id,
    String? firstNameAr,
    String? fatherNameAr,
    String? grandfatherNameAr,
    String? familyNameAr,
    String? phone,
    String? legacyNationalNo,
    int? regionId,
    String? digitalIdNumber,
    String? photoPath,
  }) =>
      Citizen(
        id: id ?? this.id,
        firstNameAr: firstNameAr ?? this.firstNameAr,
        fatherNameAr: fatherNameAr ?? this.fatherNameAr,
        grandfatherNameAr: grandfatherNameAr ?? this.grandfatherNameAr,
        familyNameAr: familyNameAr ?? this.familyNameAr,
        phone: phone ?? this.phone,
        legacyNationalNo: legacyNationalNo ?? this.legacyNationalNo,
        regionId: regionId ?? this.regionId,
        digitalIdNumber: digitalIdNumber ?? this.digitalIdNumber,
        photoPath: photoPath ?? this.photoPath,
      );
}

File: apps/api-dotnet/Data/Entities/Citizen.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarh.Api.Data.Entities;

[Table("citizens")]
public class Citizen
{
    [Column("id")] public Guid Id { get; set; }

    [Column("first_name_ar")] public string FirstNameAr { get; set; } = "";
    [Column("father_name_ar")] public string FatherNameAr { get; set; } = "";
    [Column("grandfather_name_ar")] public string GrandfatherNameAr { get; set; } = "";
    [Column("family_name_ar")] public string FamilyNameAr { get; set; } = "";

    [Column("full_name_ar")] public string FullNameAr { get; set; } = "";

    [Column("first_name_en")] public string? FirstNameEn { get; set; }
    [Column("father_name_en")] public string? FatherNameEn { get; set; }
    [Column("grandfather_name_en")] public string? GrandfatherNameEn { get; set; }
    [Column("family_name_en")] public string? FamilyNameEn { get; set; }

    [Column("mother_name_ar")] public string? MotherNameAr { get; set; }
    [Column("legacy_national_no")] public string? LegacyNationalNo { get; set; }
    [Column("family_book_no")] public string? FamilyBookNo { get; set; }

    [Column("gender")] public string Gender { get; set; } = "";
    [Column("birth_date")] public DateTime BirthDate { get; set; }
    [Column("birth_place")] public string? BirthPlace { get; set; }
    [Column("nationality")] public string? Nationality { get; set; }
    [Column("marital_status")] public string? MaritalStatus { get; set; }

    [Column("phone")] public string? Phone { get; set; }
    [Column("email")] public string? Email { get; set; }

    [Column("region_id")] public int? RegionId { get; set; }
    [Column("municipality_id")] public int? MunicipalityId { get; set; }
    [Column("address_ar")] public string? AddressAr { get; set; }

    [Column("photo_path")] public string? PhotoPath { get; set; }
    [Column("signature_path")] public string? SignaturePath { get; set; }

    [Column("created_by")] public Guid? CreatedBy { get; set; }
    [Column("auth_user_id")] public Guid? AuthUserId { get; set; }
    [Column("is_active")] public bool IsActive { get; set; }

    [Column("created_at")] public DateTimeOffset CreatedAt { get; set; }
    [Column("updated_at")] public DateTimeOffset UpdatedAt { get; set; }
}
